#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"

WorldInfo {
  contactProperties [
    ContactProperties {
      coulombFriction [
        -1
      ]
    }
  ]
}
Viewpoint {
  orientation -0.5287400026121013 -0.06494079674794909 0.8462958717585103 3.023320748990821
  position 0.7194969732340695 -0.4562462552340405 1.6194559406890754
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 2 2
}
Robot {
  translation 0.13949 -0.629553 -0.00404113
  rotation -0.007382616407926979 0.0035311382818954884 0.9999665134580294 -1.5641753071795863
  children [
    Gyro {
    }
    DEF caster_2 BallJoint {
      jointParameters BallJointParameters {
        position 9.414860506734557
        anchor -0.024 -0.03 0.007
      }
      jointParameters2 JointParameters {
        position 0.9926282550988654
      }
      jointParameters3 JointParameters {
        position 4.285084832872586
        axis 0 1 0
      }
      endPoint Solid {
        translation -0.024067478807226265 -0.03002420266313558 0.007051873041089126
        rotation 0.6678017236040811 -0.2433489804944635 0.7034359470795644 4.489909409177696
        children [
          DEF caster Shape {
            appearance PBRAppearance {
            }
            geometry Sphere {
              radius 0.006
            }
          }
        ]
        name "solid(2)"
        boundingObject USE caster
        physics Physics {
        }
      }
    }
    DistanceSensor {
      translation 0.014 -0.03 0.025
      rotation 0 0 1 -0.9646003061004252
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.005 0.005 0.005
          }
          castShadows FALSE
        }
      ]
      name "IR_L1"
      physics Physics {
      }
    }
    DistanceSensor {
      translation -0.013 -0.028 0.025
      rotation 0 0 1 -2.0117996938995746
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.005 0.005 0.005
          }
          castShadows FALSE
        }
      ]
      name "IR_R1"
      physics Physics {
      }
    }
    DistanceSensor {
      translation 0.038 -0.02 0.025
      rotation 0 0 1 -1.3
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.005 0.005 0.005
          }
          castShadows FALSE
        }
      ]
      name "IR_L2"
      physics Physics {
      }
    }
    DistanceSensor {
      translation -0.035 -0.02 0.025
      rotation 0 0 1 -1.7595903061004252
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.005 0.005 0.005
          }
          castShadows FALSE
        }
      ]
      name "IR_R2"
      physics Physics {
      }
    }
    DEF caster_1 BallJoint {
      jointParameters BallJointParameters {
        position 15.807992610823913
        anchor 0.024 -0.03 0.007
      }
      jointParameters2 JointParameters {
        position 0.5919774333329983
      }
      jointParameters3 JointParameters {
        position -6.136436872578638
        axis 0 1 0
      }
      endPoint Solid {
        translation 0.023996671158990814 -0.029976721977864412 0.006936276925359931
        rotation -0.6041406068295981 -0.17766027752399632 0.7768210559515373 4.498304673851007
        children [
          DEF caster Shape {
            appearance PBRAppearance {
            }
            geometry Sphere {
              radius 0.006
            }
          }
        ]
        name "solid(3)"
        boundingObject USE caster
        physics Physics {
        }
      }
    }
    DEF Body Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
          url [
            "../../../../../../Downloads/Downloads/mouse texture/micromouse_Material.001_BaseColor.png"
          ]
        }
        roughnessMap ImageTexture {
          url [
            "../../../../../../Downloads/Downloads/mouse texture/micromouse_Material.001_Roughness.png"
          ]
        }
        metalnessMap ImageTexture {
          url [
            "../../../../../../Downloads/Downloads/mouse texture/micromouse_Material.001_Metallic.png"
          ]
        }
        normalMap ImageTexture {
          url [
            "../../../../../../Downloads/Downloads/mouse texture/micromouse_Material.001_Normal.png"
          ]
        }
      }
      geometry Mesh {
        url [
          "../../../../../../Downloads/Downloads/micromouse.obj"
        ]
      }
    }
    DEF left_wheel HingeJoint {
      jointParameters HingeJointParameters {
        position 48.662540622812024
        anchor 0.04 0.011 0.018
      }
      device [
        PositionSensor {
          name "left wheel encoder"
        }
        RotationalMotor {
          name "left wheel"
        }
      ]
      endPoint Solid {
        translation 0.040065853649281086 0.01101568925422658 0.017821150229567033
        rotation 0.6869622796974452 0.23779964272863946 0.6866834468596288 2.6749530769298704
        children [
          DEF left_wheel Shape {
            appearance PBRAppearance {
              baseColor 0.1 0.1 0.1
              metalness 0
            }
            geometry Cylinder {
              height 0.015
              radius 0.015
            }
          }
        ]
        boundingObject USE left_wheel
        physics Physics {
          mass 1
        }
      }
    }
    DEF right_wheel HingeJoint {
      jointParameters HingeJointParameters {
        position 46.64804682388427
        anchor -0.04 0.011 0.018
      }
      device [
        PositionSensor {
          name "right wheel encoder"
        }
        RotationalMotor {
          name "right wheel"
        }
      ]
      endPoint Solid {
        translation -0.0400968591155364 0.010996493047364162 0.01812517372001382
        rotation 0.6979433853894145 -0.17356403795911157 0.6948025298734103 3.483802223445279
        children [
          DEF left_wheel Shape {
            appearance PBRAppearance {
              baseColor 0.1 0.1 0.1
              metalness 0
            }
            geometry Cylinder {
              height 0.015
              radius 0.015
            }
          }
        ]
        name "solid(1)"
        boundingObject USE left_wheel
        physics Physics {
          mass 1
        }
        rotationStep 0.01
      }
    }
  ]
  boundingObject Mesh {
    url [
      "../../../../../../Downloads/Downloads/collosion.stl"
    ]
  }
  physics Physics {
    damping Damping {
      linear 0
      angular 0
    }
  }
  controller "custom_robot_controller"
}
